org.eventb.emf.persistence/src/org/eventb/emf/persistence/SaveResourcesCommand.java

```diff
@@ -0,0 +1,111 @@
+/*******************************************************************************
+ *  Copyright (c) 2017 University of Southampton.
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v1.0
+ *  which accompanies this distribution, and is available at
+ *  http://www.eclipse.org/legal/epl-v10.html
+ *
+ *  Contributors:
+ *  University of Southampton - Initial implementation
+ *******************************************************************************/
+package org.eventb.emf.persistence;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.eclipse.core.runtime.IAdaptable;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.emf.transaction.Transaction;
+import org.eclipse.emf.transaction.TransactionalEditingDomain;
+import org.eclipse.emf.workspace.AbstractEMFOperation;
+
+/**
+ * Save resources EMF transactional command. This command can be used to save
+ * multiple Event-B EMF models into Rodin resources. It wraps the saves in an
+ * EMF transaction (this is useful because the save may modify attributes of the
+ * source model)
+ *
+ * Resources must all be in the given editing domain's resource set and must be
+ * marked as modified. If no collection of resources is passed, all modified
+ * resources will be saved.
+ * 
+ *
```

Comment:
Author information (or who should be blamed) :D

Reply:
ok added in 4ca47c2